test/e2e/storage/vsphere_volume_cluster_ds.go

```diff
@@ -0,0 +1,111 @@
+package storage
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+	Tests to verify volume provisioning on a clustered datastore
+	1. Static provisioning
+	2. Dynamic provisioning
+	3. Dynamic provisioning with spbm policy
+
+	This test reads env
+	1. CLUSTER_DATASTORE which should be set to clustered datastore
+	2. VSPHERE_SPBM_POLICY_DS_CLUSTER which should be set to a tag based spbm policy tagged to a clustered datastore
+*/
+var _ = SIGDescribe("Volume Provision On Clustered Datastore", func() {
+	f := framework.NewDefaultFramework("volume-provision")
+
+	var client clientset.Interface
+	var namespace string
+	clusterDatastore := os.Getenv("CLUSTER_DATASTORE")
+	Expect(clusterDatastore).NotTo(BeEmpty())
+	var scParameters map[string]string
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+	})
+
+	/*
+		Steps:
+		1. Create volume options with datastore to be a clustered datastore
+		2. Create a vsphere volume
+		3. Create podspec with volume path. Create a corresponding pod
+		4. Verify disk is attached
+		5. Delete the pod and wait for the disk to be detached
+	*/
+
+	It("verify static provision", func() {
+		var volumePath string
+		vsp, err := vsphere.GetVSphere()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a test vsphere volume")
+		volumeOptions := new(vclib.VolumeOptions)
```

Comment:
Can you use `createVSphereVolume(vsp *vsphere.VSphere, volumeOptions *vclib.VolumeOptions) (string, error)`.

This is available in `vsphere_utils.go`